code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplate.kt

```diff
@@ -41,276 +40,184 @@ internal sealed class AEPPushTemplate {
-        )
-        expandedBodyTextColor = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.EXPANDED_BODY_TEXT_COLOR, null
-        )
-        titleTextColor = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.TITLE_TEXT_COLOR, null
-        )
-        smallIconColor = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.SMALL_ICON_COLOR, null
-        )
-        notificationBackgroundColor = DataReader.optString(
-            data,
-            PushTemplateConstants.PushPayloadKeys.NOTIFICATION_BACKGROUND_COLOR,
-            null
-        )
-        isFromIntent = false
+    init {
+        // extract the notification payload version
+        payloadVersion = data.getString(PushPayloadKeys.VERSION)?.toInt()
+            ?: throw IllegalArgumentException("Required field \"${PushPayloadKeys.VERSION}\" not found.")
+
+        // extract the text information
+        title = data.getString(PushPayloadKeys.TITLE)
+            ?: throw IllegalArgumentException("Required field \"${PushPayloadKeys.TITLE}\" not found.")
+        body = data.getString(PushPayloadKeys.BODY)
+            ?: throw IllegalArgumentException("Required field \"${PushPayloadKeys.BODY}\" not found.")
+        expandedBodyText = data.getString(PushPayloadKeys.EXPANDED_BODY_TEXT)
+        ticker = data.getString(PushPayloadKeys.TICKER)
+
+        // extract the template type
+        templateType = PushTemplateType.fromString(data.getString(PushPayloadKeys.TEMPLATE_TYPE))
+        isFromIntent = data is IntentData
+
+        // extract the basic media information
+        imageUrl = data.getString(PushPayloadKeys.IMAGE_URL)
+
+        // extract the action information
+        actionUri = data.getString(PushPayloadKeys.ACTION_URI)
+        actionType = ActionType.valueOf(data.getString(PushPayloadKeys.ACTION_TYPE) ?: ActionType.NONE.name)
+
+        // extract the icon information
+        smallIcon = data.getString(PushPayloadKeys.SMALL_ICON) ?: data.getString(PushPayloadKeys.LEGACY_SMALL_ICON)
+        largeIcon = data.getString(PushPayloadKeys.LARGE_ICON)
+
+        // extract the color components
+        titleTextColor = data.getString(PushPayloadKeys.TITLE_TEXT_COLOR)
+        bodyTextColor = data.getString(PushPayloadKeys.BODY_TEXT_COLOR)
+        backgroundColor = data.getString(PushPayloadKeys.BACKGROUND_COLOR)
+        smallIconColor = data.getString(PushPayloadKeys.SMALL_ICON_COLOR)
+
+        // extract the other notification properties
+        tag = data.getString(PushPayloadKeys.TAG)
+        sound = data.getString(PushPayloadKeys.SOUND)
+        channelId = data.getString(PushPayloadKeys.CHANNEL_ID)
+        badgeCount = data.getInteger(PushPayloadKeys.BADGE_COUNT) ?: 0
+        isNotificationSticky = data.getBoolean(PushPayloadKeys.STICKY)
+
+        // extract notification priority and visibility
+        priorityString = data.getString(PushPayloadKeys.PRIORITY)
```

Comment:
We needed to map the priority to importance for API level > N right?

Reply:
you are right.
I added a new method getNotificationImportance() that requires Android N and above. Which translates the priorityString to Importance Int